File: 409-longest-palindrome/longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        if (s.size() == 0) return 0;
        int total = 0;

        unordered_set<char> seen = {};
        for (char c : s){
            if (seen.count(c)){
                total += 2;
                seen.erase(c);
            }else{
                seen.insert(c);
            }
        }

        if (seen.size()){
            total += 1;
            // Can put 1 in middle
        }
        return total;
    }
};

File: 409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        counts = {}
        for c in s:
            counts[c] = counts.get(c, 0) + 1
        
        total = 0
        middle_exists = False
        for c in counts:
            if counts[c] % 2 == 1:
                counts[c] -= 1
                middle_exists = True
            
            total += counts[c]

        return total + int(middle_exists)